### PR TITLE
Add examples of proxying ping

### LINKED ISSUE #2094: Separate ping from dashboard?
### Do you want to request a *feature* or report a *bug*?
Feature (or security bug, see it as you will)

* Enable `[web]` or `--web`, and anything that can reach the health check at `/ping` has full dashboard access. 
* Disable it, and no way to have a health check

Assume I have a reasonable defense-in-depth network: Internet -> public subnet (DMZ) -> private. I put a load balancer (e.g. AWS ELB) in my pubic subnet, which, in turn, has access to my private (where traefik runs). 

I want my load balancer to use the `/ping` health check to see the liveness of traefik. But if I do, then my public subnet (and ELB) has complete access to my dashboard. If I remove it, I cannot check traefik's health.

I should be able to do one of several things:

* Define a route in traefik on its usual entrypoints that is a health check, e.g. `http://:80/ping` even though `http://:80/` normally serves all of my proxied traffic
* Define a separate port for dashboard and `/ping`, so I can expose my health check port comfortably without exposing my dashboard

### PR DESCRIPTION
**Pull request by deitch**
via regular entrypoint and dedicated entrypoint.

### Description

Add examples of how to expose `/ping` on one of:

* a regular entrypoint
* a dedicated port entrypoint

This is distinct from running it on the normal admin port (default to `:8080`) and allows you to expose `/ping` to other networks without exposing the admin port.

As requested by @ldez .

Closes #2094

### CODE REVIEW

File: docs/user-guide/examples.md [hunk 1]
```diff
@@ -296,3 +296,76 @@ defaultEntryPoints = ["http"]
 IdleTimeout = "360s"
 ProvidersThrottleDuration = "5s"
 ```
+
+## Securing Ping Health Check
+The `/ping` health-check URL is enabled together with the web admin panel, enabled with the command-line `--web` or config file option `[web]`. Thus, if you have a regular path for `/foo` and an entrypoint on `:80`, you would access them as follows:
```
Comment (ldez) [inline docs/user-guide/examples.md]: Could add an empty line between the before this line?
Could you put one sentence by line?

File: docs/user-guide/examples.md [hunk 2]
```diff
@@ -296,3 +296,76 @@ defaultEntryPoints = ["http"]
...
+
+* Regular path: `http://hostname:80/foo`
+* Admin panel: `http://hostname:8080/`
+* Ping URL: `http://hostname:8080/ping`
+
+However, for security reasons, you may want to be able to expose the `/ping` health-check URL to outside health-checkers, e.g. an Internet service or cloud load-balancer, _without_ exposing your admin panel's port. In many environments, the security staff may not _allow_ you to expose it.
```
Comment (ldez) [inline docs/user-guide/examples.md]: Could you put one sentence by line?

File: docs/user-guide/examples.md [hunk 3]
```diff
@@ -296,3 +296,76 @@ defaultEntryPoints = ["http"]
...
+
+You have two options:
+
+* Enable `/ping` on a regular entrypoint
+* Enable `/ping` on a dedicated port
+
+### Enable ping health check on a regular entrypoint
+
+To proxy `/ping` from a regular entrypoint to the admin one without exposing the panel, do the following:
+
```
Comment (ldez) [inline docs/user-guide/examples.md]: Could you remove extra empty line?

File: docs/user-guide/examples.md [hunk 4]
```diff
@@ -296,3 +296,76 @@ defaultEntryPoints = ["http"]
...
+  [frontends.traefikadmin]
+  backend = "traefik"
+    [frontends.traefikadmin.routes.ping]
+    rule = "Path:/ping"
+```
+
+The above creates a new backend called `traefik`, lisenting on `http://localhost:8080`, i.e. the local admin port. We only expose the admin panel via the `frontend` named `traefikadmin`, and only expose the `/ping` Path.
+
+## Enable ping health check on dedicated port
+If you do not want to or cannot expose the health-check on a regular entrypoint - e.g. your security rules do not allow it, or you have a conflicting path - then you can enable health-check on its own entrypoint. Use the following config:
```
Comment (ldez) [inline docs/user-guide/examples.md]: Could add an empty line between the before this line?
Could you put one sentence by line?

File: docs/user-guide/examples.md [hunk 5]
```diff
@@ -296,3 +296,76 @@ defaultEntryPoints = ["http"]
...
+
+In the above example, you would access a regular path, admin panel and health-check as follows:
+
+* Regular path: `http://hostname:80/foo`
+* Admin panel: `http://hostname:8080/`
+* Ping URL: `http://hostname:8082/ping`
+
+Note the unique port for `/ping`.
+
+In the above example, it is _very_ important that you create a named dedicated entrypoint, and do **not** include it in `defaultEntryPoints`. Otherwise, you are likely to expose _all_ services via that entrypoint.
```
Comment (ldez) [inline docs/user-guide/examples.md]: Could you put one sentence by line?
And could you add empty line at the end of this file?

File: docs/user-guide/examples.md [hunk 6]
```diff
@@ -296,3 +296,76 @@ defaultEntryPoints = ["http"]
...
+
+The above is similar to the previous example, but instead of enabling `/ping` on the _default_ entrypoint, we enable it on a _dedicated_ entrypoint.
+
+In the above example, you would access a regular path, admin panel and health-check as follows:
+
+* Regular path: `http://hostname:80/foo`
+* Admin panel: `http://hostname:8080/`
+* Ping URL: `http://hostname:8082/ping`
+
+Note the unique port for `/ping`.
```
Comment (ldez) [inline docs/user-guide/examples.md]: Could you replace `unique` by `dedicated`

File: docs/user-guide/examples.md (hunk 1 above)
Comment (deitch) [reply docs/user-guide/examples.md]: Sure, updating now. Out of curiosity, why? I have never seen markdown done with a CRLF between each sentence.

**Comment by deitch** [comment]
Updated. Back to you.

File: docs/user-guide/examples.md (hunk 1 above)
Comment (ldez) [reply docs/user-guide/examples.md]: In Markdown, `\n` between two sentences is replaced by a simple ` ` (Expect in Github 😉 )

https://daringfireball.net/projects/markdown/syntax#p

File: docs/user-guide/examples.md [hunk 7]
```diff
@@ -296,3 +296,83 @@ defaultEntryPoints = ["http"]
...
+  [frontends.traefikadmin]
+  backend = "traefik"
+    [frontends.traefikadmin.routes.ping]
+    rule = "Path:/ping"
+```
+
+The above creates a new backend called `traefik`, lisenting on `http://localhost:8080`, i.e. the local admin port.
+We only expose the admin panel via the `frontend` named `traefikadmin`, and only expose the `/ping` Path.
+
+## Enable ping health check on dedicated port
```
Comment (ldez) [inline docs/user-guide/examples.md]: `##` -> `###`

Comment (deitch) [reply docs/user-guide/examples.md]: Updated

**Comment by deitch** [comment]
Uh oh, needs rebase...

**Comment by deitch** [comment]
Rebased

**Comment by deitch** [comment]
@ldez are there any outstanding changes requested? I think we got them all?

File: docs/user-guide/examples.md [hunk 8]
```diff
@@ -300,3 +300,82 @@ providersThrottleDuration = "5s"
...
+
+The above is similar to the previous example, but instead of enabling `/ping` on the _default_ entrypoint, we enable it on a _dedicated_ entrypoint.
+
+In the above example, you would access a regular path, admin panel and health-check as follows:
+
+* Regular path: `http://hostname:80/foo`
+* Admin panel: `http://hostname:8080/`
+* Ping URL: `http://hostname:8082/ping`
+
+Note the dedicated port for `/ping`.
```
Comment (nmengin) [inline docs/user-guide/examples.md]: Maybe adding the port number in the note?

File: docs/user-guide/examples.md [hunk 9]
```diff
@@ -300,3 +300,82 @@ providersThrottleDuration = "5s"
...
+
+In the above example, it is _very_ important that you create a named dedicated entrypoint, and do **not** include it in `defaultEntryPoints`.
```
Comment (nmengin) [inline docs/user-guide/examples.md]: s/that you create a named dedicated entrypoint/to create a dedicated entrypoint/.

>  and do **not** include it in `defaultEntryPoints`

I don't understand what cannot be included in the `defaultEntryPoint`. Can you develop?

File: docs/user-guide/examples.md [hunk 10]
```diff
@@ -300,3 +300,82 @@ providersThrottleDuration = "5s"
...
+Otherwise, you are likely to expose _all_ services via that entrypoint.
```
Comment (nmengin) [inline docs/user-guide/examples.md]: Do you want to say that if you don't specify a rule on the frontend (`rule = "Path:/ping"`) you can expose all the web admin paths through the `ping` entrypoint?

**Comment by deitch** [comment]
Thanks @nmengin ; updating now.

**Comment by deitch** [comment]
Updated.

**Pull request merged**